pycbc/types/array_cpu.py

```diff
@@ -93,7 +102,62 @@ def vdot(self, other):
 def squared_norm(self):
     """ Return the elementwise squared norm of the array """
     return (self.data.real**2 + self.data.imag**2)
-    
+
+_ALLOWED_DTYPES = [_np.float32, _np.float64, _np.complex64,
+                   _np.complex128, _np.uint32]
+_ALLOWED_SCALARS = [int, long, float, complex] + _ALLOWED_DTYPES
+
```

Comment:
Are these different than those set in pycbc.types.Array? Can we just import that directly, so we don't have to redefine it?